src/map/skill.cpp

```diff
@@ -9826,10 +9826,17 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 		break;
 
 	case BS_GREED:
-		if(sd){
-			clif_skill_nodamage(src,*bl,skill_id,skill_lv);
-			map_foreachinallrange(skill_greed,bl,
-				skill_get_splash(skill_id, skill_lv),BL_ITEM,bl);
+		if (sd != nullptr) {
+#ifdef	RENEWAL
+			if (pc_inventoryblank(sd) >= 5) { // Greed skill should be disabled if the character have less than 5 slots [Haydrich]
+				clif_skill_nodamage(src, *bl, skill_id, skill_lv, 1);
+				map_foreachinallrange(skill_greed, bl, skill_get_splash(skill_id, skill_lv), BL_ITEM, bl);
+			} else
+				clif_msg_color(sd, MSI_SKILL_INVENTORY_KINDCNT_OVER, color_table[COLOR_RED]);
+#else
+			clif_skill_nodamage(src, *bl, skill_id, skill_lv);
+			map_foreachinallrange(skill_greed, bl, skill_get_splash(skill_id, skill_lv), BL_ITEM, bl);
+#endif
```

Comment:
```suggestion
			if (pc_inventoryblank(sd) < 5) { // Greed skill should be disabled if the character have less than 5 slots [Haydrich]
				clif_msg_color(sd, MSI_SKILL_INVENTORY_KINDCNT_OVER, color_table[COLOR_RED]);
				break;
			}
#endif
			clif_skill_nodamage(src, *bl, skill_id, skill_lv);
			map_foreachinallrange(skill_greed, bl, skill_get_splash(skill_id, skill_lv), BL_ITEM, bl);
```
This removes the need to duplicate the item pickup code. This doesn't take in to account the comment Daegaladh left though!

Reply:
Yes i was preparing to cleanup it but thanks, but i guess we need to add else for the MSI_CANT_GET_ITEM_BECAUSE_COUNT on pre-re